src/VecSim/algorithms/svs/svs_tiered.h

```diff
@@ -0,0 +1,756 @@
+                // we hold it until the iterator is depleted or freed.
+                acquire_svs_iterator();
+                auto cur_svs_results = svs_iterator->getNextResults(n_res, BY_SCORE_THEN_ID);
+                svs_code = cur_svs_results->code;
+                svs_results.swap(cur_svs_results->results);
+                VecSimQueryReply_Free(cur_svs_results);
+                handle_svs_depletion();
+            } else {
+                if (flat_results.size() < n_res && !flat_iterator->isDepleted()) {
+                    auto tail = flat_iterator->getNextResults(n_res - flat_results.size(),
+                                                              BY_SCORE_THEN_ID);
+                    flat_results.insert(flat_results.end(), tail->results.begin(),
+                                        tail->results.end());
+                    VecSimQueryReply_Free(tail);
+                }
+
+                while (svs_results.size() < n_res && svs_iterator != depleted() &&
+                       svs_code == VecSim_OK) {
+                    auto tail =
+                        svs_iterator->getNextResults(n_res - svs_results.size(), BY_SCORE_THEN_ID);
+                    svs_code =
+                        tail->code; // Set the svs_results code to the last `getNextResults` code.
+                    // New batch may contain better results than the previous batch, so we need to
+                    // merge. We don't expect duplications (hence the <false>), as the iterator
+                    // guarantees that no result is returned twice.
+                    VecSimQueryResultContainer cur_svs_results(this->allocator);
+                    merge_results<false>(cur_svs_results, svs_results, tail->results, n_res);
+                    VecSimQueryReply_Free(tail);
+                    svs_results.swap(cur_svs_results);
+                    filter_irrelevant_results(svs_results);
+                    handle_svs_depletion();
+                }
+            }
+
+            if (VecSim_OK != svs_code) {
+                return new VecSimQueryReply(this->allocator, svs_code);
+            }
+
+            VecSimQueryReply *batch;
+            batch = compute_current_batch(n_res);
+
+            if (order == BY_ID) {
+                sort_results_by_id(batch);
+            }
+            size_t batch_len = VecSimQueryReply_Len(batch);
+            this->updateResultsCount(batch_len);
+
+            return batch;
+        }
+
+        // DISCLAIMER: After the last batch, one of the iterators may report that it is not
+        // depleted, while all of its remaining results were already returned from the other
+        // iterator. (On single-value indexes, this can happen to the svs iterator only, on
+        // multi-value
+        //  indexes, this can happen to both iterators).
+        // The next call to `getNextResults` will return an empty batch, and then the iterators will
```

Comment:
```suggestion
        // iterator. (On single-value indexes, this can happen to the svs iterator only, on
        // multi-value indexes, this can happen to both iterators).
        // The next call to `getNextResults` will return an empty batch, and then the iterators will
```

Reply:
Thank you.
Kind of formatting artefacts.
Fixed.